src/core/model/table/typed_column_data.h

```diff
@@ -173,16 +174,20 @@ class TypedColumnDataFactory {
 
     inline static std::vector<TypeId> const kAllCandidateTypes = {
             +TypeId::kDate, +TypeId::kInt, +TypeId::kBigInt, +TypeId::kDouble, +TypeId::kString};
-    inline static std::unordered_map<TypeId, std::regex> const kTypeIdToRegex = {
-            {+TypeId::kBigInt, std::regex(R"(^(\+|-)?\d{20,}$)")},
-            {+TypeId::kInt, std::regex(R"(^(\+|-)?\d{1,19}$)")},
-            {+TypeId::kNull, std::regex(Null::kValue.data())},
-            {+TypeId::kEmpty, std::regex(R"(^$)")}};
+    inline static std::unordered_map<TypeId, boost::regex> const kTypeIdToRegex = {
+            {TypeId::kDate,
+             boost::regex(
+                     R"(^([0-9]{4})[-.\/]?(1[0-2]|0[1-9]|[1-9])[-.\/]?(3[0-1]|0[1-9]|[1-9]|[1-2][0-9])$)")},
+            {TypeId::kDouble, boost::regex(R"(^([0-9]|\+|-|\.|e|E|i|I|n|N|f|F|a|A)+$)")},
```

Comment:
Use `^[+-]?(\d+(\.\d*)?|\.\d+)([eE][+-]?\d+)?$|^[+-]?(inf|nan)$` cause this matches too many invalid numbers:
https://regex101.com/r/vV26nJ/1

Reply:
This regex doesn't support `INF`, `NAN` and hexadecimal floating-point values, which are supported by [`std::strtod`](https://en.cppreference.com/w/cpp/string/byte/strtof).

Should we use `^[+-]?(\d+(\.\d*)?|\.\d+)([eE][+-]?\d+)?$|^[+-]?(?i)(inf|nan)(?-i)$|^[+-]?0[xX](((\d|[a-f]|[A-F]))+(\.(\d|[a-f]|[A-F])*)?|\.(\d|[a-f]|[A-F])+)([pP][+-]?\d+)?$`?
https://regex101.com/r/k4ZhhJ/2
I'm not sure if it's equivalent to `std::strtod`, but if it is, then `kDoubleCheck` is not needed anymore.

Reply:
yes, we should, thanks